src/Microsoft.AspNetCore.Mvc.Core/Routing/UrlHelperFactory.cs

```diff
@@ -11,7 +13,27 @@ public class UrlHelperFactory : IUrlHelperFactory
         /// <inheritdoc />
         public IUrlHelper GetUrlHelper(ActionContext context)
         {
-            return new UrlHelper(context);
+            var httpContext = context.HttpContext;
+
+            if (httpContext == null)
+            {
+                throw new ArgumentNullException(nameof(httpContext));
+            }
+
+            if (httpContext.Items == null)
+            {
+                throw new ArgumentNullException(nameof(httpContext.Items));
```

Comment:
This is not correct. Use `ArgumentException` with FormatPropertyOfTypeCannotBeNull